ee/clickhouse/sql/session_recording_events.py

```diff
@@ -71,26 +76,53 @@
 created_at,
 _timestamp,
 _offset
-FROM {database}.kafka_{table_name}
+FROM {database}.kafka_session_recording_events
 """.format(
-    table_name=SESSION_RECORDING_EVENTS_TABLE, cluster=CLICKHOUSE_CLUSTER, database=CLICKHOUSE_DATABASE,
+    target_table="writable_session_recording_events"
+    if settings.CLICKHOUSE_REPLICATION
+    else SESSION_RECORDING_EVENTS_DATA_TABLE(),
+    cluster=settings.CLICKHOUSE_CLUSTER,
+    database=settings.CLICKHOUSE_DATABASE,
+)
+
+
+# Distributed engine tables are only created if CLICKHOUSE_REPLICATED
+
+# This table is responsible for writing to sharded_session_recording_events based on a sharding key.
+WRITABLE_SESSION_RECORDING_EVENTS_TABLE_SQL = lambda: SESSION_RECORDING_EVENTS_TABLE_BASE_SQL.format(
+    table_name="writable_session_recording_events",
+    cluster=settings.CLICKHOUSE_CLUSTER,
+    engine=Distributed(data_table=SESSION_RECORDING_EVENTS_DATA_TABLE(), sharding_key="sipHash64(distinct_id)"),
```

Comment:
If you distribute by team_id you end up with shards that are really inbalanced - it would be a bad sharding key. The 'correct' shard key for us is person_id - we'll get there as part of plans in https://github.com/PostHog/product-internal/pull/240.

However this discussion is also not in scope for this PR - my goal is to bring us into sync with how our schema works on cloud. You can also read more on this at https://posthog.com/handbook/engineering/event-ingestion.

Think of it as documenting the current state.